dpipe/predict/shape.py

```diff
@@ -111,8 +112,17 @@ def wrapper(x, *args, **kwargs):
             elif ((shape - local_size) < 0).any() or ((local_stride - shape + local_size) % local_stride).any():
                 raise ValueError('Input cannot be patched without remainder.')
 
-            if stream:
-                patches = predict(divide(x, local_size, local_stride, input_axis, get_boxes=get_boxes), *args, **kwargs)
+            if async_predict:
+                divide_wrapper = make_batch if batch_size is not None else lambda x, batch_size: x
```

Comment:
Don't forget to adapt branch with `async_predict == False` for `batch_size is not None`

Reply:
Done